src/main/java/Reporter.java

```diff
@@ -0,0 +1,103 @@
+import java.io.File;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Arrays;
+import java.io.Writer;
+import java.io.FileNotFoundException;
+
+/**
+ * @version 1
+ * @author James Ward
+ * @author Clint Hall
+ */
+public class Reporter {
+    private static final String TAB = "    ";
+
+    /**
+     * Write report to a writer.
+     * @param writer The writer.
+     * @param sb2List List of Sb2 objects which are the subject of the report.
+     */
+    public void writeReport(Writer writer, List<Sb2> sb2List) {
+        try (PrintWriter printWriter = new PrintWriter(writer)) {
+            writeReport(printWriter, sb2List);
+        }
+    }
+    /**
+     * Write report to a file.
+     * @param destPath Path to the destinatin file.
+     * @param sb2List List of Sb2 objects which are the subject of the report.
+     */
+    public void writeReport(String destPath, List<Sb2> sb2List) {
+        try (PrintWriter printWriter = new PrintWriter(new File(destPath))) {
+            writeReport(printWriter, sb2List);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        /*File reportFile = new File(destPath, sb2.getName());
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(reportFile))) {
+            bw.write("Hi, I'm a file!");
+            bw.newLine();
+        } catch (IOException e) {
+
+        }
+        return reportFile();*/
+    }
+    /**
+     * Write report to a PrintWriter.
+     * @param pw The PrintWriter.
+     * @param sb2List List of Sb2 objects which are the subject of the report.
+     */
+    public void writeReport(PrintWriter pw, List<Sb2> sb2List) {
+        pw.write("Scratch Report\n\n");
+        pw.write("Number of projects: " + sb2List.size() + "\n\n");
+        for (int i = 0; i < sb2List.size(); i++) {
+            Sb2 sb2 = sb2List.get(i);
+            reportProject(i + 1, pw, sb2List.get(i));
+        }
+    }
+    /**
+     * Report one Scratch project, represented by one Sb2 object from the list.
+     * @param projectNo The index of the project in the list.  Starting from 1.
```

Comment:
So, why are we 1-basing the project index? I guess it's more human readable, but we could easily internalize that change so that calling code doesn't need to know that they should add 1 to the index. Or is it better to ensure that they do know so that they aren't surprised when the print code increments the argument? I dunno. Anyone else have thoughts?